File: core/baker-annotations/src/main/java/com/ing/baker/recipe/annotations/FiresEvent.java
package com.ing.baker.recipe.annotations;

import javax.inject.Qualifier;
import java.lang.annotation.*;

/**
 * Annotation for an interaction to show what events can fired from this interaction
 */
@Qualifier
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface FiresEvent {
    Class<?>[] oneOf();
}

File: core/baker-annotations/src/main/kotlin/com/ing/baker/recipe/annotations/FiresEvent.kt
package com.ing.baker.recipe.annotations

import javax.inject.Qualifier
import kotlin.reflect.KClass

/**
 * Annotation for an interaction to show what events can fired from this interaction
 */
@Qualifier
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
annotation class FiresEvent(
    val oneOf: Array<KClass<*>>
)
